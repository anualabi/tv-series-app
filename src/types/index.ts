export interface SeriesProps {
    Title: string;
    Year?: string;
    Rated?: string;
    Released: string;
    Runtime?: string;
    Genre?: string;
    Director?: string;
    Writer?: string;
    Actors?: string;
    Plot: string;
    Language?: string;
    Country?: string;
    Awards?: string;
    Poster: string;
    Ratings?: [
        {
            Source: string;
            Value: string;
        }
    ],
    Metascore?: string;
    imdbRating: string;
    imdbVotes?:string;
    imdbID?: string;
    Type?: string;
    totalSeasons: string;
    Response?: string;
}

export type EpisodesProps = {
    Title: string;
    Plot: string;
    Released: string;
    Episode: string;
    Poster: string;
    imdbRating: string;
    imdbID: string;
}[]

export interface SeasonProps {
    Title: string;
    Plot?: string;
    Season: string;
    totalSeasons: string;
    Episodes: EpisodesProps,
    Response: string;
}

export interface EpisodeProps {
    Title: string;
    Year?: string;
    Rated?: string;
    Released: string;
    Season?: string;
    Episode: string;
    Runtime?: string;
    Genre?: string;
    Director?: string;
    Writer?: string;
    Actors?: string;
    Plot: string;
    Language?: string;
    Country?: string;
    Awards?: string;
    Poster: string;
    Ratings?: [
        {
            Source: string;
            Value: string;
        }
    ],
    Metascore?: string;
    imdbRating: string;
    imdbVotes?:string;
    imdbID?: string;
    seriesID?: string;
    Type?: string;
    Response?: string;
}
