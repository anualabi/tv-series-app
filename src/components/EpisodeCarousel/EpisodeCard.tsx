import React from 'react'
import Image from 'next/image';

import { EpisodeProps } from '@/types';

export type EpisodeCardProps = Pick<EpisodeProps, 'Poster' | 'Episode' | 'Title' | 'Plot'>

const EpisodeCard: React.FC<EpisodeCardProps> = ({ Poster, Episode, Title, Plot }) => {
  return (
    <div className="overflow-hidden w-56">
      <div className="relative w-[201px] h-[134px]">
        <Image
          src={Poster}
          alt={`${Title} image`}
          style={{objectFit: "cover"}}	
          className="w-full h-full"
          width={201}
          height={134}
        />
        <div className="absolute top-0 left-0 bg-white w-[30px] h-[30px] flex items-center justify-center">
          <p className="text-sm font-bold text-black">{Episode}</p>
        </div>
      </div>
      <div className="py-4 bg-opacity-50 relative w-[201px] h-[138px]">
        <h2 className="text-sm md:text-base  text-white font-bold mb-2 line-clamp-1">{Title}</h2>
        <p className='text-xs md:text-sm text-white font-light line-clamp-3'>{Plot}</p>
      </div>
    </div>
  )
}

export default EpisodeCard
