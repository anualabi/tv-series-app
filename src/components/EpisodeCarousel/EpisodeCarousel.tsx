import React from 'react'

import Carousel from '@/components/ui/Carousel';
import { EpisodesProps } from '@/types';

import EpisodeCard from './EpisodeCard';

export type EpisodeCarouselProps = {
  episodes: EpisodesProps
};

const EpisodeCarousel = ({ episodes }: EpisodeCarouselProps) => {

  const episodeList = episodes.map((episode) => (
    <EpisodeCard key={episode.imdbID} {...episode} />
  ))
  
  return (
    <div className='flex'>
      <Carousel items={episodeList} />
    </div>
  )
}

export default EpisodeCarousel;