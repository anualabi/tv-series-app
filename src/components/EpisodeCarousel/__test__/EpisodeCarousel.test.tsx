import React from 'react';
import { render } from '@testing-library/react';
import EpisodeCarousel from '../EpisodeCarousel';
import '@testing-library/jest-dom/extend-expect';
import { EpisodesProps } from '@/types';

const mockProps = [
    {
        Title: 'Better-Like',
        Plot: 'This is a sample plot for testing purposes.',
        Released: '2018-08-12',
        Episode: '5',
        Poster: 'https://test-poster1.jpg',
        imdbRating: '9.0',
        imdbID: '123'
    },
    {
        Title: 'Familiar-Like',
        Plot: 'This is a sample plot for testing purposes.',
        Released: '2018-08-19',
        Episode: '6',
        Poster: 'https://test-poster2.jpg',
        imdbRating: '8.5',
        imdbID: '456'
    },
]

const customRender = (props: EpisodesProps) => render(<EpisodeCarousel episodes={props} />);

describe('EpisodeCarousel', () => {
    test('should render snapshot', () => {
        const { asFragment } = customRender(mockProps)
        
        expect(asFragment()).toMatchSnapshot();
    });
});
