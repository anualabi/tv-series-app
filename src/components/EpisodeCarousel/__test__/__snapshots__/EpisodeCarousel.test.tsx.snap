// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeCarousel should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex"
  >
    <div
      class="h-full flex w-full overflow-x-hidden relative"
    >
      <div
        class="inline-flex gap-4"
      >
        <div
          class="overflow-hidden w-56"
        >
          <div
            class="relative w-[201px] h-[134px]"
          >
            <img
              alt="Better-Like image"
              class="w-full h-full"
              data-nimg="1"
              decoding="async"
              height="134"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Ftest-poster1.jpg&w=640&q=75"
              srcset="/_next/image?url=https%3A%2F%2Ftest-poster1.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Ftest-poster1.jpg&w=640&q=75 2x"
              style="color: transparent; object-fit: cover;"
              width="201"
            />
            <div
              class="absolute top-0 left-0 bg-white w-[30px] h-[30px] flex items-center justify-center"
            >
              <p
                class="text-sm font-bold text-black"
              >
                5
              </p>
            </div>
          </div>
          <div
            class="py-4 bg-opacity-50 relative w-[201px] h-[138px]"
          >
            <h2
              class="text-sm md:text-base  text-white font-bold mb-2 line-clamp-1"
            >
              Better-Like
            </h2>
            <p
              class="text-xs md:text-sm text-white font-light line-clamp-3"
            >
              This is a sample plot for testing purposes.
            </p>
          </div>
        </div>
        <div
          class="overflow-hidden w-56"
        >
          <div
            class="relative w-[201px] h-[134px]"
          >
            <img
              alt="Familiar-Like image"
              class="w-full h-full"
              data-nimg="1"
              decoding="async"
              height="134"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Ftest-poster2.jpg&w=640&q=75"
              srcset="/_next/image?url=https%3A%2F%2Ftest-poster2.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Ftest-poster2.jpg&w=640&q=75 2x"
              style="color: transparent; object-fit: cover;"
              width="201"
            />
            <div
              class="absolute top-0 left-0 bg-white w-[30px] h-[30px] flex items-center justify-center"
            >
              <p
                class="text-sm font-bold text-black"
              >
                6
              </p>
            </div>
          </div>
          <div
            class="py-4 bg-opacity-50 relative w-[201px] h-[138px]"
          >
            <h2
              class="text-sm md:text-base  text-white font-bold mb-2 line-clamp-1"
            >
              Familiar-Like
            </h2>
            <p
              class="text-xs md:text-sm text-white font-light line-clamp-3"
            >
              This is a sample plot for testing purposes.
            </p>
          </div>
        </div>
      </div>
      <div
        class="absolute bottom-0 right-3 hidden md:block"
      >
        <button
          class="mx-2 text-white focus:outline-none opacity-50 cursor-not-allowed"
          data-testid="left-arrow"
          disabled=""
          tabindex="0"
        >
          <svg
            height="19.414"
            viewBox="0 0 28.707 19.414"
            width="28.707"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="tail-left"
              transform="translate(29.207 19.207) rotate(-180)"
            >
              <path
                d="M0,0H28"
                fill="none"
                id="Shape"
                stroke="#fff"
                stroke-miterlimit="10"
                stroke-width="1"
                transform="translate(0.5 9.5)"
              />
              <path
                d="M0,0,9,9,0,18"
                data-name="Shape"
                fill="none"
                id="Shape-2"
                stroke="#fff"
                stroke-linecap="square"
                stroke-miterlimit="10"
                stroke-width="1"
                transform="translate(19.5 0.5)"
              />
            </g>
          </svg>
        </button>
        <button
          class="mx-2 text-white focus:outline-none "
          data-testid="right-arrow"
          tabindex="0"
        >
          <svg
            height="19.414"
            viewBox="0 0 28.707 19.414"
            width="28.707"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="tail-right"
              transform="translate(-0.5 0.207)"
            >
              <path
                d="M0,0H28"
                fill="none"
                id="Shape"
                stroke="#fff"
                stroke-miterlimit="10"
                stroke-width="1"
                transform="translate(0.5 9.5)"
              />
              <path
                d="M0,0,9,9,0,18"
                data-name="Shape"
                fill="none"
                id="Shape-2"
                stroke="#fff"
                stroke-linecap="square"
                stroke-miterlimit="10"
                stroke-width="1"
                transform="translate(19.5 0.5)"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
